Source/SwiftLintFramework/Protocols/Rule.swift

```diff
@@ -8,10 +8,19 @@
 
 import SourceKittenFramework
 
+public enum RuleKind: String {
+    case lint
+    case idiomatic
+    case style
+    case metrics
+    case performance
+}
+
 public protocol Rule {
     init() // Rules need to be able to be initialized with default values
     init(configuration: Any) throws
     static var description: RuleDescription { get }
+    static var kind: RuleKind { get }
```

Comment:
What about using an option set? I think the advantage is that a rule can be classifed with multiple "labels" without those labels having to be protocols. I like the idea of protocols as labels too. 
The emum restriction kind of reminds me of  http://www.jessesquires.com/enums-as-configs/